cmake/modules/SwiftSource.cmake

```diff
@@ -274,6 +280,7 @@ function(_compile_swift_files dependency_target_out_var_name)
   set(module_file)
   set(module_doc_file)
 
+  set(module_command)
   if(NOT SWIFTFILE_IS_MAIN)
```

Comment:
It was actually just a move as the previous code had use adding '-emit-module ... -parse-as-library' on one branch and '-parse-as-library' on the other.  But I agree that this is getting a little unwieldy at its current size.  Thats probably worth someone refactoring a whole bunch of stuff though, which is better left to who is going to own it long term.